docs/getting_started/how_to_build_an_api.md

```diff
@@ -0,0 +1,215 @@
+# How to build an API
+
+The purpose of Gosoline is to help with API building. Much of its functionality is API related, and now we will cover the most important parts.
+
+## Package apiserver
+
+An apiserver is, at its bare minimum, a module that runs indefinitely, listens to a port for requests, and provides answers to those requests. 
+
+Package apiserver provides a convenient way to create API server objects:
+
+[embedmd]:# (../../pkg/apiserver/server.go /func New\(/ /\)/)
+```go
+func New(definer Definer)
+```
+
+Where _Definer_ is a method which returns a _*Definitions_:
+
+[embedmd]:# (../../pkg/apiserver/definition.go /type Definer/ /\)\n/)
+```go
+type Definer func(ctx context.Context, config cfg.Config, logger log.Logger) (*Definitions, error)
+```
+
+Lastly, this is how _Definitions_ looks like:
+
+[embedmd]:# (../../pkg/apiserver/definition.go /type Definitions struct/ /\n}/)
+```go
+type Definitions struct {
```

Comment:
All fields are unexported... so why does a user of gosoline care about them?

Reply:
If a framework tells me about inner workings I don't care about (but then also doesn't explain them), I would only get the feeling that the framework is badly designed (why do I need to know about them) or documented (why does the documentation not know what private and public means)